File: owner/src/main/java/lk/quadrate/owner/entity/Owner.java
package lk.quadrate.owner.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Owner {

    @Id
    @SequenceGenerator(
            name = "owner_sequence",
            sequenceName = "owner_sequence",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "owner_sequence"
    )
    private Long id;
    @Column(unique = true)
    private String userId;
    @NotBlank
    @Column(nullable = false)
    private String firstName;
    @NotBlank
    @Column(nullable = false)
    private String lastName;
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Gender gender;
    @Column(unique = true)
    @NotBlank
    private String governmentId;
    private String countryCode;
    private String contactNumber;
    private String address;
    private String city;
    private String country;
    private String postalCode;
    private LocalDateTime dateOfBirth;
    private LocalDateTime createdAt = LocalDateTime.now();
    private Boolean isProfileCompleted = false;

}

File: angular-frontend/src/app/Dashboard/Models/owner.ts
export class Owner {
  id?:number;
  userId?:string;
  firstName?:string;
  lastName?:string;
  gender?:string;
  governmentId?:string;
  countryCode?:string;
  contactNumber?:string
  address?:string;
  city?:string;
  country?:string;
  postalCode?:string;
  dateOfBirth?:string;
  createdAt?:string;
  isProfileCompleted?:boolean;


  constructor(id: number, userId: string, firstName: string, lastName: string, gender: string, governmentId: string, countryCode: string, contactNumber: string, address: string, city: string, country: string, postalCode: string, dateOfBirth: string, createdAt: string, isProfileCompleted: boolean) {
    this.id = id;
    this.userId = userId;
    this.firstName = firstName;
    this.lastName = lastName;
    this.gender = gender;
    this.governmentId = governmentId;
    this.countryCode = countryCode;
    this.contactNumber = contactNumber;
    this.address = address;
    this.city = city;
    this.country = country;
    this.postalCode = postalCode;
    this.dateOfBirth = dateOfBirth;
    this.createdAt = createdAt;
    this.isProfileCompleted = isProfileCompleted;
  }
}
